soc/arc/snps_arc_hsdk4xd/soc_ctrl.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Synopsys, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef _ARC_HSDK4XD_SOC_CTRL_H_
+#define _ARC_HSDK4XD_SOC_CTRL_H_
+
+#ifdef _ASMLANGUAGE
+.macro soc_early_asm_init_percpu
+	/* case of HS4XD cpu */
+	lr	 r0, [_ARC_V2_LPB_BUILD]
+	breq r0, 0, skip_lpb_setup
```

Comment:
As this is code for one exact SoC which is ASIC (so it won't be changed as FPGA based SoCs) - we can drop BCR check here.